source/runtime_gui.cpp

```diff
@@ -4370,4 +4384,27 @@ void reshade::runtime::destroy_imgui_resources()
 	_imgui_pipeline_layout = {};
 }
 
+bool reshade::runtime::activate_overlay(bool activate, reshade::api::input_source source)
+{
+	const bool requested = activate;
+
+	reshade::invoke_addon_event<reshade::addon_event::reshade_overlay_activation>(
```

Comment:
This ideally has protection against an add-on calling `activate_overlay`/`open_overlay` from within the `reshade_overlay_activation`/`reshade_open_overlay` event and also cannot be called when building ReShade without add-on support (also using the suggestion of using the return value to skip instead of a parameter):
```c++
bool reshade::runtime::open_overlay(bool open, api::input_source source)
{
#if RESHADE_ADDON
	if (!_is_in_api_call)
	{
		_is_in_api_call = true;
		const bool skip = invoke_addon_event<addon_event::reshade_open_overlay>(this, &open, source);
		_is_in_api_call = false;
		if (skip)
			return false;
	}
#endif

	_show_overlay = open;

	if (open)
		_imgui_context->NavInputSource = static_cast<ImGuiInputSource>(source);

	return true;
}
```